hnn_core/network.py

```diff
@@ -764,6 +799,235 @@ def _get_src_type_and_pos(self, gid):
+        key = f'gbar_L5Pyr_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'ampa'
+        self._all_to_all_connect(
+            src_cell, target_cell, loc, receptor, nc_dict)
+
+        src_cell = 'L2Pyr'
+        key = f'gbar_L2Pyr_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'ampa'
+        self._all_to_all_connect(
+            src_cell, target_cell, loc, receptor, nc_dict)
+
+    def _all_to_all_connect(self, src_cell, target_cell,
+                            loc, receptor, nc_dict,
+                            allow_autapses=True, unique=False):
+        """Generate connectivity list given lists of sources and targets.
+
+        Parameters
+        ----------
+        src_cell : str
+            Source cell type.
+        target_cell : str
+            Target cell type.
+        loc : str
+            If 'proximal' or 'distal', the corresponding
+            dendritic sections from Cell.sect_loc['proximal']
+            or Cell.Sect_loc['distal'] are used
+        receptor : str
+            The receptor.
+        nc_dict : dict
+            The connection dictionary containing keys
+            A_delay, A_weight, lamtha, and threshold.
+        allow_autapses : bool
+            If True, allow connecting neuron to itself.
+        unique : bool
+            If True, each target cell gets one "unique" feed.
+            If False, all src_type cells are connected to
+            all target_type cells.
+        """
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        target_gids = self.gid_ranges[_long_name(target_cell)]
+
+        src_start = src_gids[0]  # Necessary for unique feeds
+        for target_gid in target_gids:
+            if unique:
+                src_gids = [target_gid + src_start]
+            for src_gid in src_gids:
+                if not allow_autapses and src_gid == target_gid:
+                    continue
+                self.add_connection(
+                    src_gid, target_gid, loc, receptor,
+                    nc_dict['A_weight'], nc_dict['lamtha'])
+
+    def add_connection(self, src_gid, target_gid, loc, receptor,
+                       weight, lamtha):
+        """Appends connections to connectivity list
+
+        Parameters
+        ----------
+        src_gid : int
+            Integer identifying source cell ID.
+        target_gid : list | int
+            Integer or list of integers identifying target cell ID.
+        loc : str
+            Location of synapse on target cell. Must be
+            'proximal' or 'distal'.
+        receptor : str
+            Synaptic receptor of connection. Must be one of:
+            'ampa', 'nmda', 'gabaa', or 'gabab'.
+        weight : float
+            Synaptic weight on target cell.
+        lamtha : float
+            Space constant.
+        """
+        conn = dict()
+        threshold = self.threshold
```

Comment:
this overrides the threshold in the param file?

Reply:
Yes that's correct, I plan to modify it so that network instantiation sets the attribute from the param file.